haskell: add all haskell-mode keybindings to literate-haskell-mode

jb55 opened the pull request:
Rebased and cleaned up version of #3459 by @codygman

Closes #3459
Fixes #3458

Few more things to do

- [x] Test literate haskell + ghc-mod
- [x] Test literate haskell + hindent
- [x] Add `literate-haskell-mode` to `spacemacs/declare-prefix-for-mode`

StreakyCobra commented:
Will also fix #3458

jb55 commented:
kk updated description :+1:

robbyoconnor commented:
This is sexy as hell -- nice work @jb55 :purple_heart: :yellow_heart: :heart: :blue_heart: :green_heart:

d12frosted commented:
Nice, @jb55 thanks for PR :100:

StreakyCobra commented on layers/+lang/haskell/packages.el:
```diff
@@ -29,12 +29,15 @@
+        (spacemacs/set-leader-keys-for-major-mode m "hf" 'helm-hoogle)))))
 
 (defun haskell/post-init-flycheck ()
   (spacemacs/add-flycheck-hook 'haskell-mode))
```
Just to be sure, flycheck doesn't support literate-haskell?

jb55 replied:
hmm it should, I'm going to be using this a lot tomorrow so I'll be able to test it a bunch

jb55 replied:
I noticed the hlint checker having issues, had to disable it. haskell-ghc seems to work fine. I'll update this line in a bit.

StreakyCobra commented:
Lines 53 + 56-66: just to be sure, ghc-mode doesn't support literate-haskell?
Lines 122-128: I think the prefixes' names should also be added in literate-haskell.
Lines 274-275: just to be sure, hindent doesn't support literate-haskell?

codygman commented:
Thanks @jb55!

@StreakyCobra I don't know offhand, I just know that opening a literate haskell file didn't work at the time I wrote this.

jb55 commented:
@StreakyCobra I don't use hindent or ghc-mod, but now I have an excuse to try them out. I've updated the description to track the progress.

syl20bnr commented:
:+1: 
Remove [WIP] when you judge it is ready.

jb55 commented:
tested and rebased :+1: 

* couldn't get ghc-mod working, so I'll leave it as is for now 
* hindent doesn't work in literate-haskell-mode
* added prefixes

d12frosted commented:
> couldn't get ghc-mod working, so I'll leave it as is for now

I thought ghc-mod supports lhs.

jb55 commented:
ok then I'll have to take your word on it since I wasn't able to test it. I've added its bindings to literate-haskell-mode.

That should be it...

syl20bnr commented:
Thank you :+1: 
Cherry-picked into develop branch, you can safely delete your branch.
